wallet/src/wallet.rs

```diff
@@ -138,6 +138,27 @@ pub trait CapeWalletExt<'a, Backend: CapeWalletBackend<'a> + Sync + 'a> {
         // We may return a `WrapReceipt`, i.e., a record commitment to track wraps, once it's defined.
     ) -> Result<(), CapeWalletError>;
 
+    /// Construct the information required to wrap an asset, but do not submit a transaction.
+    ///
+    /// A new record opening is created and returned. This function will not invoke the contract.
+    /// For a version which does, use [CapeWalletExt::wrap].
+    async fn build_wrap(
+        &mut self,
+        cap_asset: AssetDefinition,
+        dst_addr: UserAddress,
+        amount: u64,
+    ) -> Result<RecordOpening, CapeWalletError>;
+
+    /// Submit a wrap transaction to the CAPE contract.
+    ///
+    /// `ro` should be the record opening returned by `build_wrap`.
+    async fn submit_wrap(
+        &mut self,
+        src_addr: EthereumAddr,
+        cap_asset: AssetDefinition,
```

Comment:
Why do we need the `AssetDefinition`? It should always be the same as `ro.asset_def` right?

Reply:
Good catch! In `submitwrap` I actually got the `AssetDefinition` from `ro.asset_def` and passed it to `wallet.submit_wrap`, so yeah, I can just pass `ro` to `wallet.submit_wrap`.